src/lib/awsConnectionParameters.ts

```diff
@@ -282,26 +282,78 @@ export async function getCredentials(awsParams: AWSConnectionParameters): Promis
 async function queryRegionFromMetadata(): Promise<string> {
     // SDK doesn't support discovery of region from EC2 instance metadata, so do it manually. We set
     // a timeout so that if the build host isn't EC2, we don't hang forever
-    return new Promise<string>((resolve, reject) => {
-        const metadataService = new AWS.MetadataService()
-        metadataService.httpOptions.timeout = 5000
-        metadataService.request('/latest/dynamic/instance-identity/document', (err, data) => {
-            try {
+    const imdsOptions = { httpOptions: { timeout: 5000 }, maxRetries: 2 }
+
+    const metadataService = new AWS.MetadataService(imdsOptions)
+
+    let token
```

Comment:
This eliminates the `token as string` on line 298

```suggestion
    let token: string
```